Allow combined type libraries to be combined

2shortplanks opened the pull request:
This pull request allows combined type libraries to themselves be combined with other type libraries

2shortplanks committed:
Allow combined type libraries to be combined

This commit allows combined type libraries to themselves be combined with other type libraries

karenetheridge commented:
Some documentation updates would be helpful -- I don't really understand what you're doing here.

2shortplanks commented:
So, this means just like you can combine multiple type libraries into one combined type library, you can also combine combined type libraries together with type libraries and other combined type libraries to create uber combined type libraries.

i'll write more documentation tomorrow to make this clear

2shortplanks committed:
Improve documentation

Add more documentation and make it clear that you can build combined
type libraries out of combined type libraries as well as type libraries

2shortplanks commented:
@karenetheridge bump

autarch commented on lib/MooseX/Types/Combine.pm:
```diff
@@ -22,21 +22,64 @@ use namespace::autoclean;
     my ($class, @types) = @_;
     my $caller = caller;
 
+    my $where_to_import_to = $caller;
```
Might as well not bother assigning to `$caller` at all any more, right? Does the variable get used anywhere else?

2shortplanks replied:
Line 93 in the error message.

autarch replied:
Ah, ok

autarch commented on lib/MooseX/Types/Combine.pm:
```diff
@@ -108,6 +152,13 @@ sub _provided_types {
     %$types;
 }
 
+sub type_names {
```
This is a new public method without documentation. Should it be private or documented? I'm inclined to say it should be documented, since it seems like a nice addition.

2shortplanks replied:
Documented in follow up commit

autarch reviewed (changes requested):
This mostly looks good to me except for a few issues.

2shortplanks committed:
document type_names method

karenetheridge commented:
hooray for über type libraries!  thanks for this; will release imminently.
